### PR TITLE
CorfuStore: Stream tails per table api needed for search

### PR DESCRIPTION
**Pull request by hisundar**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

**Review by zfrenette — APPROVED** [review]

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java [hunk 1]
```diff
@@ -293,6 +293,14 @@ public int count() {
         return corfuTable.keySet();
     }
 
+    /**
+     * Return the address of the latest updated made in this table.
+     *
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/runtime/collections/Table.java:298]: Missing @return in javadoc

**Review by WenbinZhu — APPROVED** [review]

**Review by Lujie1996 — APPROVED** [review]

File: test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java [hunk 2]
```diff
@@ -104,6 +109,58 @@ public void checkDirtyReads() throws Exception {
         log.debug(table.getMetrics().toString());
     }
 
+    @Test
+    public void checkIndexes() throws Exception {
```
Comment (pankti-m) [inline test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java:113]: can we rename the test to something more explanatory?  something like testSecondaryIndexes()

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java (hunk 1 above)
Comment (hisundar) [reply runtime/src/main/java/org/corfudb/runtime/collections/Table.java:298]: added, thanks

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java [hunk 3]
```diff
@@ -293,6 +293,14 @@ public int count() {
         return corfuTable.keySet();
     }
 
+    /**
+     * Return the address of the latest updated made in this table.
```
[the remark this answers is not in the document]
Comment (hisundar) [reply runtime/src/main/java/org/corfudb/runtime/collections/Table.java:297]: fixed, thanks

File: runtime/src/main/java/org/corfudb/runtime/collections/Table.java [hunk 4]
```diff
@@ -293,6 +293,14 @@ public int count() {
...
+     *
+     */
+    public long getHighestSequence() {
```
[the remark this answers is not in the document]
Comment (hisundar) [reply runtime/src/main/java/org/corfudb/runtime/collections/Table.java:300]: We can, but I am strictly following the ask, that's why I put it here. we do expose the table directly anyway in UFO

File: test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java (hunk 2 above)
Comment (hisundar) [reply test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java:113]: done

**Review by pankti-m — APPROVED** [review]

**Pull request merged**